### DISCUSSION TITLE
Why do clustered_index has an option at all?

### DISCUSSION THREAD

**Discussion by midnight-wonderer** [mongodb/mongo-ruby-driver#2729] [Q&A]
Hey,
I am interested in clustered collections for key-value use cases; BTW, I came from Mongoid.

In Mongoid's document (+ some info on this repo), I can specify

~~~Ruby
store_in collection_options: {
  clustered_index: {
    key: { _id: 1 },
    unique: true,
  }
}
~~~

in my model; I supposed.

Which is weird because
~~~Ruby
{
  key: { _id: 1 },
  unique: true,
}
~~~
is the only possible value; they can't be anything else.

Why do we have options for `key` and `unique` at all?

**Comment by alexbevi** [comment]
Hey @midnight-wonderer,

> Why do we have options for key and unique at all?

Just to clarify, the `clustered_index` document is an option [passed to the `create` command](https://www.mongodb.com/docs/manual/reference/command/create/#std-label-create.clusteredIndex) to indicate to the MongoDB Server a [Clustered Collection](https://www.mongodb.com/docs/manual/core/clustered-collections/#clustered-collections) is being created.

Though no other values for `key` and `unique` are presently supported, the Driver's API is being designed to support future changes to the Server's command options.
